src/dialects/sqlite/query.js

```diff
@@ -241,14 +236,16 @@ export class SqliteQuery extends AbstractQuery {
 
           // cannot use arrow function here because the function is bound to the statement
           function afterExecute(executionError, results) {
-            try {
-              complete();
-              // `this` is passed from sqlite, we have no control over this.
-
-              resolve(query._handleQueryResponse(this, columnTypes, executionError, results, errForStack.stack));
-            } catch (error) {
-              reject(error);
+            if (executionError) {
+              executionError.sql = query.sql;
+              const rethrowErrStack = new Error(executionError.message);
+              rethrowErrStack.stack = `${rethrowErrStack.stack.split('\n').slice(0, 2).join('\n')}\n${errForStack.stack}`;
+              reject(query.formatError(executionError, rethrowErrStack.stack));
             }
+
+            complete();
```

Comment:
This change should be reverted. `complete` must remain in a try-catch, and there is no need to re-create an Error here